src/compiler/program.ts

```diff
@@ -1376,7 +1376,7 @@ namespace ts {
         }
 
         function getTypeChecker() {
-            return noDiagnosticsTypeChecker || (noDiagnosticsTypeChecker = createTypeChecker(program, /*produceDiagnostics:*/ false));
+            return noDiagnosticsTypeChecker || (noDiagnosticsTypeChecker = diagnosticsProducingTypeChecker || createTypeChecker(program, /*produceDiagnostics:*/ false));

```

Comment:
no.. no.. no.. There is potential that the diagnostic producing checker was initialized and later was thrown away because there was some kind of cancellation. That makes it even harder to guarantee you have correct type checker or same type checker.

Reply:
`runWithCancellationToken` resets both anyway.

Reply:
`dropDiagnosticsProducingTypeChecker` is the only thing that could otherwise prematurely drop the checker; but despite it's presence in our internal API, it's never actually called anywhere (vestigial API?)

Reply:
`dropDiagnosticsProducingTypeChecker` is now gone (it was unused) and `runWithCancellationToken` specifically resets both to avoid a situation like this, so I think this is fine now.